articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,331 @@
+- [Loops in Dart](#loops-in-dart)
+   - [For Loop](#for-loop)
+   - [While Loop](#while-loop)
+   - [Do-While Loop](#do-while-loop)
+- [Other Key Concepts](#other-key-concepts)
+   - [Break Keyword](#break-keyword) 
+   - [Continue Keyword](#continue-keyword) 
+- [Applications of Loops](#applications-of-loops)
+- [Further Reading](#further-reading)
+- [Conclusion ](#conclusion)
+
+### General Loop Structure
+
+There are two types of loops:
+1. #### Definite Loops
+These are loops that we know the number of times we want to execute the code.***For example***, The `For Loop`, later on through this article, we shall come to understand that it has a definite number of iterations known to the programmer.
+
+2. #### Indefinite Loops
+These refer to loops we do not know the number of times we want to execute the code. ***For example***, The `While Loop` and The `Do-While Loop`, later on, we shall come to understand what it means by the iterations not known to the user.
+
+The general loop structure refers to the fundamental components of loops in all languages and usage. All the loop types mentioned above have the following characeristics:
+
+- *Counter Variable* - We also know this as the initializer. It is the starting point of the loop where it executes. 
+
+- *Condition Check* - This is the condition a loop has to start or execute. After it's met, it ends.
+
+- *Increment or Decrement Counter Variable*  - This refers to the number in which counter variable increases or decreases until the loop meets the condition.
+
+
+### Loops in Dart
+We should note that the syntax of Dart Loops is like the ones in Java Programming Language.
+
+There are three loops in Dart:
+
+1. #### For Loop
+
+For Loop is an example of [Definite Loops](#definite-loops).
+
+>Syntax of For Loop.
+
+```Dart
+void main(){
+
+   for (initializer; condition; increment/decrement){
+  
+   //put your code here;
+
+}
+
+}
+
+```
+
+>How the For Loop works.
+
+The flow of the For Loop: 
+
+**Initialize - Condition Check - Code Execute - Increment**.
+
+The counter variable initializes, then it checks the condition if it's true the code executes, then the counter variable increases depending on the value of increment. The increased counter variable initializes, and it re-checks the condition. If it's still true, the code executes and so on until the condition check is false or not met, then the loop ends.
+
+
+***Example:***
+
+```Dart
+
+void main(){
+for(var i = 0; i<4; i++){
+    print ("Hello");
+ 
+ }
+
+}
+
+/** Output : Hello
+             Hello
+             Hello
+             Hello **/
+
+```
+Here the value of the initializer of the counter variable will increase by one from 0 based on the increment specified. If the value of i becomes equal to four or more, the loop terminates/ends since the condition states that the value of i should be less than 4. The loop is to print hello if the condition is true, hence the output is hello written four times (0,1,2,3) < 4. 
```

Comment:
Here the value ~~of the initializer~~ of the counter variable ... 

You can just say the counter variable.